src/main/scala/cromwell/util/docker/DockerHash.scala

```diff
@@ -0,0 +1,42 @@
+package cromwell.util.docker
+
+import javax.xml.bind.DatatypeConverter
+
+import cromwell.util.StringUtil._
+
+case class DockerHash(hashType: String, hashString: String) {
+  validate()
+
+  private def validate() = {
+    DatatypeConverter.parseHexBinary(hashString)
+    val length = hashString.length
+    require(length == 8 || length == 32 || length == 64, s"unexpected hash length: $length")
+  }
+}
+
+object DockerHash {
```

Comment:
You have a couple of generator functions in here plus that validate method - would it make sense to use a proper validation here?